### PR TITLE
Change HTTP request to axios

### PR DESCRIPTION
**Pull request by cssuh**
A customer reported that proxy was working for adding account, but they were not able to perform any migration steps, so updating our requests back to axios.

### CODE REVIEW

File: extensions/azurecore/src/azureResource/utils.ts [hunk 1]
```diff
@@ -463,7 +461,6 @@ export async function makeHttpRequest<B>(
 	requestHeaders: Record<string, string> = {}
 ): Promise<AzureRestResponse<B>> {
 	const result: AzureRestResponse<B> = { response: undefined, errors: [] };
-	const httpClient: HttpClient = getProxyEnabledHttpClient();
```
Comment (cheenamalhotra) [inline extensions/azurecore/src/azureResource/utils.ts:466]: Finally clean this up as well?

Comment (cheenamalhotra) [reply extensions/azurecore/src/azureResource/utils.ts:466]: I also think this was referenced elsewhere but not actually used.. so please clean up.

**Review by cheenamalhotra — APPROVED** [review]
Other than the comments, looks good to me.

File: extensions/azurecore/src/azureResource/utils.ts [hunk 2]
```diff
@@ -527,26 +523,22 @@ export async function makeHttpRequest<B>(
 		requestUrl = `${account.properties.providerSettings.settings.armResource.endpoint}${path}`;
 	}
 
-	let response: AzureNetworkResponse<B | ErrorResponseBodyWithError> | undefined = undefined;
+	let response: AxiosResponse<B | ErrorResponseBodyWithError> | undefined;
```
Comment (Charles-Gagnon) [inline extensions/azurecore/src/azureResource/utils.ts:526]: We shouldn't need `AzureNetworkResponse` anymore, right? Make sure you go check where that's used and clean it up as necessary as well.

File: extensions/azurecore/src/azurecore.d.ts [hunk 3]
```diff
@@ -273,17 +274,6 @@ declare module 'azurecore' {
 		PATCH
 	}
 
-	/**
-	 * Custom version of NetworkResponse from @azure\msal-common\dist\network\NetworkManager.d.ts
-	 * with body renamed to data to avoid breaking changes with extensions. See
-	 * https://github.com/microsoft/azuredatastudio/pull/22761 for details.
```
Comment (Charles-Gagnon) [inline extensions/azurecore/src/azurecore.d.ts:279]: You've verified that everything is still good with the new response type?

**Review by Charles-Gagnon — APPROVED** [review]
Thanks for cleaning this all up!

File: extensions/azurecore/src/azurecore.d.ts (hunk 3 above)
Comment (cssuh) [reply extensions/azurecore/src/azurecore.d.ts:279]: Yep, I went through the migration flow and everything worked successfully for me.

**Pull request merged**